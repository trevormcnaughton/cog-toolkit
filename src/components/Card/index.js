import React from 'react'
import theme from '../../util/theme'
import styled from 'styled-components'

const Body = styled.section`
  color: ${theme.color.greyDarkest};
  width: 100%;
  padding: 1rem;
`
const Inner = styled.section`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
`
const Header = styled.header`
  display: flex;
  width: 100%;
  padding: 1rem;
  border-bottom: 1px solid ${theme.color.greyLight};
`
const Title = styled.h2`
  margin: 0;
  font-size: 1rem;
  font-weight: 900;
  color: ${theme.color.greyDark};
`
const Description = styled.p`
  margin: 0;
  word-break: break-word;
`
const Footer = styled.section`
  display: flex;
  align-items: center;
  width: 100%;
  margin-top: auto;
  padding: 1rem;
  font-weight: 600;
  font-size: 0.875rem;
`

const Card = ({
  className,
  children,
  interactable,
  title,
  style,
  description,
  footer,
}) => {
  const Card = styled.section`
    position: relative;
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    height: auto;
    color: ${theme.color.greyDarkest};
    background: #ffffff;
    border: 1px solid rgba(0, 0, 0, 0.08);
    box-shadow: 0 2px 8px 0 rgba(0, 0, 0, 0.04);
    border-radius: 4px;
    overflow: hidden;
    transition: box-shadow 200ms;
  `

  return (
    <Card className={className} style={style}>
      {children}
    </Card>
  )
}

Card.Body = Body
Card.Header = Header
Card.Title = Title
Card.Description = Description
Card.Footer = Footer

export default Card
