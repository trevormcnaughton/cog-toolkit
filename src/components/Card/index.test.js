import React from 'react'
import renderer from 'react-test-renderer'

import Card from './index'

describe('Card', () => {
  describe('snapshots', () => {
    it('renders a card component', () => {
      const tree = renderer.create(<Card>Card</Card>)

      expect(tree).toMatchSnapshot()
    })

    it('renders a card component with body', () => {
      const tree = renderer.create(
        <Card>
          <Card.Body>
            <p>Some stuff in here</p>
          </Card.Body>
        </Card>
      )

      expect(tree).toMatchSnapshot()
    })

    it('renders a card component with header, title, and body', () => {
      const tree = renderer.create(
        <Card>
          <Card.Header><Card.Title>Card Title</Card.Title></Card.Header>
          <Card.Body>
            <p>Some stuff in here</p>
          </Card.Body>
        </Card>
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
