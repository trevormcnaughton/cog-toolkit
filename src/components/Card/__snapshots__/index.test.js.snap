// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshots renders a card component 1`] = `
<section
  className="sc-bZQynM fGIniy"
  style={undefined}
>
  Card
</section>
`;

exports[`Card snapshots renders a card component with body 1`] = `
<section
  className="sc-gzVnrw hBKTKg"
  style={undefined}
>
  <section
    className="sc-bdVaJa HwIbV"
  >
    <p>
      Some stuff in here
    </p>
  </section>
</section>
`;

exports[`Card snapshots renders a card component with header, title, and body 1`] = `
<section
  className="sc-htoDjs erVDdY"
  style={undefined}
>
  <header
    className="sc-htpNat byljKJ"
  >
    <h2
      className="sc-bxivhb gbvAzg"
    >
      Card Title
    </h2>
  </header>
  <section
    className="sc-bdVaJa HwIbV"
  >
    <p>
      Some stuff in here
    </p>
  </section>
</section>
`;
